## Import Data and Library

### Import Library

In [2]:
import numpy as np
import pandas as pd
import seaborn as sns
from matplotlib import pyplot as plt
from zipfile import ZipFile

### Import Data

In [113]:
with ZipFile('big-startup-secsees-fail-dataset-from-crunchbase.zip') as raw_zip:
    raw_zip.extractall(path='datasets')
    
    raw_data = pd.read_csv('datasets/big_startup_secsees_dataset.csv')

In [18]:
raw_data.head()

permalink                    name  \
0                  /organization/-fame                   #fame   
1               /organization/-qounter                :Qounter   
2  /organization/-the-one-of-them-inc-  (THE) ONE of THEM,Inc.   
3                /organization/0-6-com                 0-6.com   
4       /organization/004-technologies        004 Technologies   

                        homepage_url  \
0                 http://livfame.com   
1             http://www.qounter.com   
2                http://oneofthem.jp   
3                 http://www.0-6.com   
4  http://004gmbh.de/en/004-interact   

                                       category_list funding_total_usd  \
0                                              Media          10000000   
1  Application Platforms|Real Time|Social Network...            700000   
2                                  Apps|Games|Mobile           3406878   
3                                        Curated Web           2000000   
4                                           Software                 -   

      status country_code state_code                 region           city  \
0  operating          IND         16                 Mumbai         Mumbai   
1  operating          USA         DE             DE - Other  Delaware City   
2  operating          NaN        NaN                    NaN            NaN   
3  operating          CHN         22                Beijing        Beijing   
4  operating          USA         IL  Springfield, Illinois      Champaign   

   funding_rounds  founded_at first_funding_at last_funding_at  
0               1         NaN       2015-01-05      2015-01-05  
1               2  2014-09-04       2014-03-01      2014-10-14  
2               1         NaN       2014-01-30      2014-01-30  
3               1  2007-01-01       2008-03-19      2008-03-19  
4               1  2010-01-01       2014-07-24      2014-07-24

## Data Inspection

### Total Amount of Rows and Columns

In [26]:
raw_data.shape

(66368, 14)

### Data Fields

In [21]:
raw_data.columns

Index(['permalink', 'name', 'homepage_url', 'category_list',
       'funding_total_usd', 'status', 'country_code', 'state_code', 'region',
       'city', 'funding_rounds', 'founded_at', 'first_funding_at',
       'last_funding_at'],
      dtype='object')

#### Data Fields Information
1. permalink: Link to Organization
2. name: Company Name
3. homepage_url: Startup Site
4. category_list: Field of company
5. funding_total_usd: Total Funding in USD
6. status: Operating Status
7. country_code: Country Code
8. state_code: State Code of company location
9. region: Region of company location
10. city: Cit of company location
11. funding_rounds: the number of times a startup opts to raise capital from the market.
12. founded_at: the date when the startup is founded
13. first_funding_at: the date when the startup get first funding
14. last_funding_at: the date when the startup get last funding

### Check Uniqeness of Each Columns

In [33]:
raw_data.nunique()

permalink            66368
name                 66102
homepage_url         61191
category_list        27296
funding_total_usd    18896
status                   4
country_code           137
state_code             311
region                1092
city                  5111
funding_rounds          19
founded_at            3978
first_funding_at      4817
last_funding_at       4518
dtype: int64

The permalink column is completely unique (it act like an id), so we can remove it later from the data.

### Missing Values

In [123]:
raw_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 66368 entries, 0 to 66367
Data columns (total 14 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   permalink          66368 non-null  object
 1   name               66367 non-null  object
 2   homepage_url       61310 non-null  object
 3   category_list      63220 non-null  object
 4   funding_total_usd  66368 non-null  object
 5   status             66368 non-null  object
 6   country_code       59410 non-null  object
 7   state_code         57821 non-null  object
 8   region             58338 non-null  object
 9   city               58340 non-null  object
 10  funding_rounds     66368 non-null  int64 
 11  founded_at         51147 non-null  object
 12  first_funding_at   66344 non-null  object
 13  last_funding_at    66368 non-null  object
dtypes: int64(1), object(13)
memory usage: 7.1+ MB


In [122]:
raw_data

permalink  \
0                                    /organization/-fame   
1                                 /organization/-qounter   
2                    /organization/-the-one-of-them-inc-   
3                                  /organization/0-6-com   
4                         /organization/004-technologies   
...                                                  ...   
66363  /organization/zznode-science-and-technology-co...   
66364                          /organization/zzzzapp-com   
66365                                /organization/Áeron   
66366                              /organization/Ôasys-2   
66367  /organization/İnovatiff-reklam-ve-tanıtım-hizm...   

                                             name  \
0                                           #fame   
1                                        :Qounter   
2                          (THE) ONE of THEM,Inc.   
3                                         0-6.com   
4                                004 Technologies   
...                                           ...   
66363               ZZNode Science and Technology   
66364                       Zzzzapp Wireless ltd.   
66365                                       ÁERON   
66366                                       Ôasys   
66367  İnovatiff Reklam ve Tanıtım Hizmetleri Tic   

                            homepage_url  \
0                     http://livfame.com   
1                 http://www.qounter.com   
2                    http://oneofthem.jp   
3                     http://www.0-6.com   
4      http://004gmbh.de/en/004-interact   
...                                  ...   
66363              http://www.zznode.com   
66364             http://www.zzzzapp.com   
66365               http://www.aeron.hu/   
66366               http://www.oasys.io/   
66367               http://inovatiff.com   

                                           category_list funding_total_usd  \
0                                                  Media          10000000   
1      Application Platforms|Real Time|Social Network...            700000   
2                                      Apps|Games|Mobile           3406878   
3                                            Curated Web           2000000   
4                                               Software                 -   
...                                                  ...               ...   
66363                                Enterprise Software           1587301   
66364        Advertising|Mobile|Web Development|Wireless            114304   
66365                                                NaN                 -   
66366  Consumer Electronics|Internet of Things|Teleco...             18192   
66367                 Consumer Goods|E-Commerce|Internet             14851   

          status country_code state_code                 region  \
0      operating          IND         16                 Mumbai   
1      operating          USA         DE             DE - Other   
2      operating          NaN        NaN                    NaN   
3      operating          CHN         22                Beijing   
4      operating          USA         IL  Springfield, Illinois   
...          ...          ...        ...                    ...   
66363  operating          CHN         22                Beijing   
66364  operating          HRV         15                  Split   
66365  operating          NaN        NaN                    NaN   
66366  operating          USA         CA            SF Bay Area   
66367  operating          NaN        NaN                    NaN   

                city  funding_rounds  founded_at first_funding_at  \
0             Mumbai               1         NaN       2015-01-05   
1      Delaware City               2  2014-09-04       2014-03-01   
2                NaN               1         NaN       2014-01-30   
3            Beijing               1  2007-01-01       2008-03-19   
4          Champaign               1  2010-01-01       2014-07-24   
... 

### Check Duplicate Startup Name

In [118]:
check_name = raw_data['name'].value_counts()

check_name[check_name > 1]

Blink     4
Peach     4
Spire     4
Roost     4
Karma     4
         ..
Stream    2
Shift     2
Splore    2
Euclid    2
Weave     2
Name: name, Length: 239, dtype: int64

There are 239 duplicated startup name, lets check wheter the startup has the same name or the data is duplicated.

In [119]:
duplicate_name = check_name[check_name > 1].index

duplicate_inspect = raw_data[raw_data['name'].str.fullmatch('|'.join(duplicate_name)) == True]

duplicate_inspect[['permalink', 'name', 'country_code', 'city', 'founded_at', 'last_funding_at']].sort_values(by='name')

permalink     name country_code  \
281                    /organization/3divaz-2   3DIVAZ          NaN   
282                    /organization/3divaz-3   3DIVAZ          CHE   
3943                    /organization/aps-llc      APS          USA   
3942                        /organization/aps      APS          NaN   
3982                        /organization/apx      APX          NLD   
...                                       ...      ...          ...   
66029                 /organization/zingbox-2  ZingBox          NaN   
17442                     /organization/eegeo    eeGeo          GBR   
30620                    /organization/kimble    eeGeo          GBR   
49053                      /organization/rumr     rumr          USA   
49054  /organization/rumr-anonymous-messenger     rumr          USA   

              city  founded_at last_funding_at  
281            NaN  2014-07-01      2015-11-01  
282        Wildegg  2014-07-01      2015-11-01  
3943       Griffin  2011-03-31      2011-12-29  
3942           NaN         NaN      2009-09-16  
3982     Amsterdam  1999-01-01      2007-12-21  
...            ...         ...             ...  
66029          NaN         NaN      2014-12-12  
17442       Dundee         NaN      2012-07-17  
30620       Dundee         NaN      2012-07-11  
49053     Brooklyn  2013-08-01      2013-12-03  
49054  Los Angeles  2013-10-01      2015-09-21  

[502 rows x 6 columns]

In [120]:
for column in duplicate_inspect.columns[2:]:
    duplicate_inspect = duplicate_inspect.drop_duplicates(subset=['name', column])

Lets check the duplicate name again

In [121]:
duplicate_inspect['name'].value_counts()

Shake        2
Leap         2
Ventrix      2
Clover       2
Bridge       2
            ..
Guru         1
Gusto        1
Handshake    1
Haven        1
ZingBox      1
Name: name, Length: 238, dtype: int64

The duplicated startup name still exist! Lets assume those startup coincidentally have the same name. Lets remove the duplicated data from raw_data.